### PR TITLE
Allow environment-specific authorizer keys

### LINKED ISSUE #2159: separate operator keys based on environment
right now we just have a single list of keys in `services/payments/authorize.go` - we should break this out per environment

### PR DESCRIPTION
**Pull request by Sneagan**
### Summary

Allow environment-specific authorizer keys. closes #2159

### Type of Change

- [x] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before Requesting Review

- [x] Does your code build cleanly without any errors or warnings?
- [x] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [x] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security and/or privacy review if needed
- [x] Have you performed a self review of this PR?

### CODE REVIEW

**Comment by Sneagan** [comment]
Relies on https://github.com/brave-intl/payment-ops/pull/48

File: services/payments/authorize.go [hunk 1]
```diff
@@ -5,27 +5,33 @@ import (
...
-	"ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEY/3VGKsrH5dp3mK5PJIHVkUMWpsmUhZkrLuZTf7Sqr kdenhartog+settlement+dev@brave.com",
+var validAuthorizerKeys = map[string][]string{
+	"production": {},
+	"staging": {
+		// @evq
+		"ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIA91/jZI+hcisdAURdqgdAKyetA4b2mVJIypfEtTyXW+ evq+settlements@brave.com",
+		// @sneagan
+		"ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDfcr9jUEu9D9lSpUnPwT1cCggCe48kZw1bJt+CXYSnh jegan+settlements@brave.com",
+	},
+	"develop": {
```
Comment (pavelbrm) [inline services/payments/authorize.go:24]: Maybe `development` would be better? Also, consistent with some other places (one example is SKUs).

Comment (Sneagan) [reply services/payments/authorize.go]: Good catch. Updated in both PRs.

**Review by evq — APPROVED** [review]
lgtm. can you delete the `Println` on line 41 while we're making changes here?

I created a follow-up issue to put an additional safeguard in place to decrease the changes of environment mismatch: https://github.com/brave-intl/bat-go/issues/2189

**Review by husobee — APPROVED** [review]

**Pull request merged**